src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -120,6 +122,28 @@ public ReceivedMessage withHook(InteractionHook hook)
         return this;
     }
 
+    private void checkIntent()
+    {
+        // Checks whether access to content is limited and the message content intent is not enabled
+        if (api != null && !contentIntentWarning && !api.isIntent(GatewayIntent.MESSAGE_CONTENT))
```

Comment:
When would `api` be null here?

Reply:
We can and should remove this from the constructor.
`new ReceivedMessage` only happens in one place, the EntityBuilder
https://github.com/DV8FromTheWorld/JDA/blob/a1c3ecdb43700139e9474f3531c1b590019a25e5/src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java#L1554

And it does so in a method that requires that `channel` be non-null.